engine/src/lib.rs

```diff
@@ -161,6 +168,14 @@ impl Engine {
 
     /// Dispatch plugin code on the given stage
     pub fn dispatch(&mut self, stage: Stage) -> Result<()> {
+        // TODO: Should this be the responsibility of something else?
+        // Pre-update formally marks the start of a new frame
+        if stage == Stage::PreUpdate {
```

Comment:
If we're planning on adding multiple stages do we want to change this from an if to a match statement?

Reply:
How about let's do that when we need to match multiple stages :sweat_smile:

Reply:
That's what the `unimplemented!()` macro is for 😎